plugins/toolkit/jetbrains-core/src/software/aws/toolkits/jetbrains/services/codewhisperer/inlay/CodeWhispererInlayManager.kt

```diff
@@ -52,19 +65,36 @@ class CodeWhispererInlayManager {
     }
 
     fun clearInlays(editor: Editor) {
-        editor.inlayModel.getInlineElementsInRange(
-            0,
-            editor.document.textLength,
-            CodeWhispererInlayInlineRenderer::class.java
-        ).forEach { disposable ->
-            Disposer.dispose(disposable)
-        }
-        editor.inlayModel.getBlockElementsInRange(
-            0,
-            editor.document.textLength,
-            CodeWhispererInlayBlockRenderer::class.java
-        ).forEach { disposable ->
-            Disposer.dispose(disposable)
+        if (!AppMode.isRemoteDevHost()) {
+            editor.inlayModel.getInlineElementsInRange(
+                0,
+                editor.document.textLength,
+                CodeWhispererInlayInlineRenderer::class.java
+            ).forEach { disposable ->
+                Disposer.dispose(disposable)
+            }
+            editor.inlayModel.getBlockElementsInRange(
+                0,
+                editor.document.textLength,
+                CodeWhispererInlayBlockRenderer::class.java
+            ).forEach { disposable ->
+                Disposer.dispose(disposable)
+            }
+        } else {
+            editor.inlayModel.getInlineElementsInRange(
+                0,
+                editor.document.textLength,
+                InlineSuffixRenderer::class.java
+            ).forEach { disposable ->
+                Disposer.dispose(disposable)
+            }
+            editor.inlayModel.getBlockElementsInRange(
+                0,
+                editor.document.textLength,
+                InlineBlockElementRenderer::class.java
```

Comment:
do we need to make sure these were all from codewhisperer

Reply:
we should, haven't figured out how, let me think a bit.